### PR TITLE
Upgrade rules_docker

### PR DESCRIPTION
**Pull request by meteorcloudy**
To fix build failure with Bazel@HEAD caused by bazelbuild/rules_docker#1533
https://buildkite.com/bazel/bazel-at-head-plus-downstream/builds/1539

### CODE REVIEW

**Comment by meteorcloudy** [comment]
FYI @aiuto

File: k8s/k8s.bzl [hunk 1]
```diff
@@ -80,9 +80,9 @@ py_library(
     if "io_bazel_rules_docker" not in excludes:
         http_archive(
             name = "io_bazel_rules_docker",
-            sha256 = "6287241e033d247e9da5ff705dd6ef526bac39ae82f3d17de1b69f8cb313f9cd",
-            strip_prefix = "rules_docker-0.14.3",
```
Comment (fejta) [inline k8s/k8s.bzl:84]: 0.14.3 the latest release, please work with rules_docker to get a new release out that includes any fixes you want.

https://github.com/bazelbuild/rules_docker/releases

**Comment by fejta** [comment]
This PR is not necessary, we auto-update to the latest rules_docker release once it is available: https://github.com/bazelbuild/rules_k8s/pull/567

**Comment by meteorcloudy** [comment]
https://github.com/bazelbuild/rules_docker/releases/tag/v0.14.4 is out, can you do a auto update for rules_docker?
Because rules_k8s is still failing with Bazel in downstream https://buildkite.com/bazel/bazel-at-head-plus-disabled/builds/679#8a53590c-7b59-450a-8053-b62047d1f881

**Comment by fejta** [comment]
https://github.com/bazelbuild/rules_k8s/pull/585 exists but is failing tests

**Pull request closed without merging**